src/test/unit_tests/test_aws_sts.js

```diff
@@ -0,0 +1,65 @@
+/* Copyright (C) 2020 NooBaa */
+'use strict';
+
+const mocha = require('mocha');
+const assert = require('assert');
+const sinon = require('sinon');
+const AWS = require('aws-sdk');
+const cloud_utils = require('../../util/cloud_utils');
+const dbg = require('../../util/debug_module')(__filename);
+const fs = require("fs");
+
+mocha.describe('AWS STS tests', function() {
+    const projectedServiceAccountToken = "/var/run/secrets/openshift/serviceaccount/oidc-token";
+    const fakeAccessKeyId = "fakeAccessKeyId";
+    const fakeSecretAccessKey = "fakeSecretAccessKey";
+    const fakeSessionToken = "fakeSessionToken";
+    const defaultSTSCredsValidity = 3600;
+    const roleArn = "arn:aws:iam::261532230807:role/noobaa_s3_sts";
+    sinon.stub(fs.promises, "readFile").withArgs(projectedServiceAccountToken)
+        .returns("web-identity-token");
+        const stsFake = {
+            assumeRoleWithWebIdentity: sinon.stub().returnsThis(),
+            promise: sinon.stub().resolves({
+                    Credentials: {
+                        AccessKeyId: fakeAccessKeyId,
+                        SecretAccessKey: fakeSecretAccessKey,
+                        SessionToken: fakeSessionToken
+                    }
+            }),
+          };
+    const STSStub = sinon.stub(AWS, 'STS').callsFake(() => stsFake);
+
+    mocha.it('should generate aws sts creds', async function() {
+        const params = {
+            aws_sts_arn: roleArn
+        };
+        const expectedParams = [
+            { RoleArn: roleArn, RoleSessionName: 'testSession', WebIdentityToken: 'web-identity-token', DurationSeconds: defaultSTSCredsValidity }
+        ];
+        const roleSessionName = "testSession";
+        const json = await cloud_utils.generate_aws_sts_creds(params, roleSessionName);
+        sinon.assert.calledOnce(STSStub);
+        sinon.assert.calledWith(stsFake.assumeRoleWithWebIdentity, ...expectedParams);
+        assert.equal(json.accessKeyId, fakeAccessKeyId);
+        assert.equal(json.secretAccessKey, fakeSecretAccessKey);
+        assert.equal(json.sessionToken, fakeSessionToken);
+        dbg.log0('test.aws.sts.assumeRoleWithWebIdentity: ', json);
+    });
+    mocha.it('should generate an STS S3 client', async function() {
+        const params = {
+            aws_sts_arn: roleArn,
+            region: 'us-east-1'
+        };
+        const additionalParams = {
+            RoleSessionName: 'testSession'
+        };
+        const s3 = await cloud_utils.createSTSS3Client(params, additionalParams);
+        dbg.log0('test.aws.sts.createSTSS3Client: ', s3);
+        assert.equal(s3.config.credentials.accessKeyId, fakeAccessKeyId);
+        assert.equal(s3.config.credentials.secretAccessKey, fakeSecretAccessKey);
+        assert.equal(s3.config.credentials.sessionToken, fakeSessionToken);
+        assert.equal(s3.config.region, 'us-east-1');
+        STSStub.restore();
```

Comment:
This restore() will not be called if this test fails before it.
The way to do it safer in mocha is to use mocha.before(...) and mocha.after(...)
See here for example:
https://github.com/ezio-auditore/noobaa-core/blob/92ec8ead6284c69badc581e9e9d1d5b7c2b624b6/src/test/unit_tests/test_namespace_fs.js#L53-L62